lisa/tools/dnsmasq.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from lisa.executable import Tool
+from lisa.tools.kill import Kill
+
+
+class Dnsmasq(Tool):
+    @property
+    def command(self) -> str:
+        return "dnsmasq"
+
+    def start(
+        self,
+        nic_name: str,
+        nic_address: str,
+        dhcp_range: str,
+    ) -> None:
+        kill = self.node.tools[Kill]
+        kill.by_name("dnsmasq")
+        # setup dnsmasq on interface `nic_name` and listen on `nic_address`
+        # assign dhcp address in `dhcp_range`
+        cmd = (
+            f"--strict-order --except-interface=lo --interface={nic_name} "
+            f"--listen-address={nic_address} --bind-interfaces "
+            f"--dhcp-range={dhcp_range} --conf-file='' "
+        )
+        cmd += f"--pid-file=/var/run/qemu-dnsmasq-{nic_name}.pid "
```

Comment:
It can be written like below, it regards like one string. You can write them in the same line. The "black" will auto format it.

```python
        cmd += f"--pid-file=/var/run/qemu-dnsmasq-{nic_name}.pid "
            f"--dhcp-leasefile=/var/run/qemu-dnsmasq-{nic_name}.leases "
            "--dhcp-no-override "
```